src/v/transform/api.cc

```diff
@@ -446,6 +448,22 @@ ss::future<> service::start() {
     co_await _batcher->start();
     co_await _manager->start();
     register_notifications();
+
+    _log_manager = std::make_unique<logging::manager<ss::lowres_clock>>(
+      _self,
+      std::make_unique<logging::rpc_client>(
+        &_rpc_client->local(), &_topic_table->local()),
+      config::shard_local_cfg().data_transforms_logging_buffer_capacity_bytes(),
+      config::shard_local_cfg().data_transforms_logging_line_max_bytes.bind(),
+      config::shard_local_cfg()
+        .data_transforms_logging_flush_interval_ms.bind());
+    auto fut = co_await ss::coroutine::as_future(_log_manager->start());
+    if (fut.failed()) {
+        vlog(tlog.warn, "Failed to start data transform log_manager");
+        // TODO(oren): might be cool to fall back to broker logs in this case.
+        // this doesn't seem like a _fatal_ error (?), but maybe stop/start
+        // transforms is a good enough recovery path
```

Comment:
Let's just make it a fatal error for now. Feels weird to have stuff silently broken.